lib/fastimage.rb

```diff
@@ -391,6 +425,11 @@ def parse_size
     send("parse_size_for_#{@type}")
   end
 
+  def parse_animated
+    @type = parse_type unless @type
+    @type == :gif ? send("parse_animated_for_#{@type}") : nil
```

Comment:
I think it should return false is the type is not gif (maybe we want to support, for instance, animated png in future).
It can return nil if the type is not found though.